addons/mail/static/src/models/message/message.js

```diff
@@ -523,6 +648,156 @@ function factory(dependencies) {
             return replace(threads);
         }
 
+        /**
+         * Apllies the hightlighting based the thread search text. 
+         * 
+         * @private
+         * @param {String} content
+         * @returns {String}
+         */
+        _highlightBasedOnThreadSearch(content) {
+            if (this.threads) {
+                for (const thread of this.threads) {
+                    if (thread.searchedText) {
+                        content = content.replace(
+                            new RegExp(`(${thread.searchedText})`,"ig"),
+                            '<b class="o_MessageInlineHighlight">$1</b>'
+                        );
+                    }
+                }
+            }
+            return content;
+        }
+
+        /**
+         * Adds Read More and Read Less togglers for the collapsable contents.
+         * 
+         * @param {String} content String rep of the DOM Content
+         * @param {Boolean} shouldOpenChidlrenRecursively 
+         * When opening a collapsable content, should it open all the children as well ?
+         * Making the effective depth of the collapsable equals 1.
+         * @returns String rep of the DOM Content processed
+         */
+        _insertMoreOrLessTogglersForCollapsableContent(content, shouldOpenChidlrenRecursively = false) {
+            /**
+             * For simplicity, we need to manipulate a DOM Element directly.
+             * Note that "unvalid" html, like nested <p> will get unested by this procedure.
+             * So the better the html conventions are respected, the better this will work.
+             */
```

Comment:
@sdegueldre I got the same result using the DOMParser. It tries to unnest it as well (and the result is not pretty...)

Reply:
I'm not getting that on either chrome or firefox, are you using the correct mimetype? (`text/xml`)

Although since we're going to end up inserting this in the page it doesn't matter all that much, it will 'unnest' at that point, I just thought I'd mention it in passing.